envoy/service/load_stats/v2/lrs.proto

```diff
@@ -59,105 +45,15 @@ service EndpointDiscoveryService {
-  //   `this issue <https://github.com/envoyproxy/envoy/issues/1285>`_ is fixed.
-  google.protobuf.UInt32Value load_balancing_weight = 4
-      [(validate.rules).uint32 = {gte: 1, lte: 128}];
-}
-
-// A group of endpoints belonging to a Locality.
-// One can have multiple LocalityLbEndpoints for a locality, but this is
-// generally only done if the different groups need to have different load
-// balancing weights or different priorities.
-message LocalityLbEndpoints {
-  // Identifies location of where the upstream hosts run.
-  Locality locality = 1;
-
-  // The group of endpoints belonging to the locality specified.
-  repeated LbEndpoint lb_endpoints = 2;
-
-  // Optional: Per priority/region/zone/sub_zone weight - range 1-128. The load
-  // balancing weight for a locality is divided by the sum of the weights of all
-  // localities  at the same priority level to produce the effective percentage
-  // of traffic for the locality.
-  //
-  // Weights must be specified for either all localities in a given priority
-  // level or none.
-  //
-  // If unspecified, each locality is presumed to have equal weight in a
-  // cluster.
-  //
-  // .. attention::
-  //
-  //   The limit of 128 is somewhat arbitrary, but is applied due to performance
-  //   concerns with the current implementation and can be removed when
-  //   `this issue <https://github.com/envoyproxy/envoy/issues/1285>`_ is fixed.
-  google.protobuf.UInt32Value load_balancing_weight = 3
-      [(validate.rules).uint32 = {gte: 1, lte: 128}];
-
-  // Optional: the priority for this LocalityLbEndpoints. If unspecified this will
-  // default to the highest priority (0).
-  //
-  // Under usual circumstances, Envoy will only select endpoints for the highest
-  // priority (0). In the event all endpoints for a particular priority are
-  // unavailable/unhealthy, Envoy will fail over to selecting endpoints for the
-  // next highest priority group.
-  //
-  // Priorities should range from 0 (highest) to N (lowest) without skipping.
-  uint32 priority = 5;
-}
-
-// [#not-implemented-hide:] Not configuration. TBD how to doc proto APIs.
-message EndpointLoadMetricStats {
-  // Name of the metric; may be empty.
-  string metric_name = 1;
-
-  // Number of calls that finished and included this metric.
-  uint64 num_requests_finished_with_metric = 2;
-
-  // Sum of metric values across all calls that finished with this metric for
-  // load_reporting_interval.
-  double total_metric_value = 3;
-}
-
 // These are stats Envoy reports to GLB every so often. Report frequency is
 // defined by
-// :ref:`LoadStatsResponse.load_reporting_interval<envoy_api_field_LoadStatsResponse.load_reporting_interval>`.
+// :ref:`LoadStatsResponse.load_reporting_interval<envoy_api_field_load_stats.LoadStatsResponse.load_reporting_interval>`.
 // Stats per upstream region/zone and optionally per subzone.
 // [#not-implemented-hide:] Not configuration. TBD how to doc proto APIs.
 message UpstreamLocalityStats {
```

Comment:
These could probably live in `envoy.api.v2.endpoint` within a file called `load_report.proto`.

Reply:
Done